### PR TITLE
cm_syns branch (new compartmental model + synapses) for review

### PR DESCRIPTION
**Pull request by pythonjam**
Hi @clinssen, @pnbabu . 
I submit my current code for review and feedback. Thanks :)

This also contains a newer compartmental model, the old PR request is therefore somewhat stale.

### CODE REVIEW

File: pynestml/cocos/co_cos_manager.py [hunk 1]
```diff
@@ -334,6 +373,8 @@ def post_symbol_table_builder_checks(cls, neuron: ASTNeuron, after_ast_rewrite:
         cls.check_variables_unique_in_scope(neuron)
         cls.check_state_variables_initialized(neuron)
         cls.check_variables_defined_before_usage(neuron, after_ast_rewrite)
+        cls.check_compartmental_model(neuron, after_ast_rewrite)
```
Comment (clinssen) [inline pynestml/cocos/co_cos_manager.py:376]: Can we run these only if code generation target is "NEST-compartmental"?

File: pynestml/codegeneration/resources_nest/ModuleClass.jinja2 [hunk 2]
```diff
@@ -65,8 +65,13 @@
 #include "{{moduleName}}.h"
 
 {% for neuron in neurons %}
+    {%- if neuron.is_compartmental_model -%}
```
Comment (clinssen) [inline pynestml/codegeneration/resources_nest/ModuleClass.jinja2:68]: Templates to their own directory according to the new scheme (https://github.com/nest/nestml/pull/669).

File: pynestml/utils/ast_synapse_information_collector.py [hunk 3]
```diff
@@ -0,0 +1,332 @@
+"""
```
Comment (clinssen) [inline pynestml/utils/ast_synapse_information_collector.py:1]: Does it make sense from an organisational perspective to have a subdirectory ``pynestml/utils/compartmental`` for these utils files?

File: tests/docstring_comment_test.py [hunk 4]
```diff
@@ -60,7 +60,10 @@ class DocstringCommentTest(unittest.TestCase):
     def test_docstring_success(self):
         self.run_docstring_test('valid')
 
-    @pytest.mark.xfail(strict=True, raises=DocstringCommentException)
+    # for some reason xfail is completely ignored when executing on my machine (python 3.8.5)
```
Comment (clinssen) [inline tests/docstring_comment_test.py:63]: Please remove this from the PR.

**Review by clinssen — CHANGES_REQUESTED** [review]
Could you make a NestCompartmentalCodeGenerator that subclasses NestCodeGenerator, so that ideally there are only minimal changes to ``nest_codegenerator.py``?

**Comment by pythonjam** [comment]
The new branch that is properly merged is cm_structured

**Pull request closed without merging**